crates/solidity/inputs/language/src/definition.rs

```diff
@@ -3336,12 +3336,29 @@ codegen_language_macros::compile!(Language(
                                     name = FunctionCallExpression,
                                     operators = [PrecedenceOperator(
                                         model = Postfix,
+                                        fields = (arguments = Required(ArgumentsDeclaration))
+                                    )]
+                                ),
+                                PrecedenceExpression(
+                                    name = CallOptionsExpression,
+                                    operators = [PrecedenceOperator(
+                                        model = Postfix,
+                                        enabled = From("0.6.2"),
+                                        error_recovery = FieldsErrorRecovery(
+                                            delimiters = FieldDelimiters(
+                                                open = open_brace,
+                                                close = close_brace,
+                                                // NOTE: Despite `NamedArguments` requiring at least one element,
+                                                // we need to disable attempting to recover from empty elements,
+                                                // because this postfix is ambiguous with `try <EXPR> {} catch {}`
+                                                // and could lead to incorrect parsing if we recover past valid syntax.
+                                                disallow_unmatched_body = true
```

Comment:
Name open for bikeshedding, but I like that the option is an explicit opt-in (with `true` value) that emphasizes that the option additionally restricts the logic to accommodate for an edge case/custom logic.

Reply:
My vote would be for `reject_partial_contents`. "Partial" here indicates the meaning better, as "unmatched" is an internal parser concept IIUC, and can change in the future. "Partial" describes it from user POV. And "Contents" because it can be many things, not to be confused with just function/contract bodies.

Reply:
As I mentioned in the chat, we always recover from a partial (prefix) match, no matter this configuration; this is about recovering if we didn't match anything at all that we expected, Maybe `reject_unrecognized_contents` would be better?

Reply:
Apologies for the confusion. Wouldn't that mean `reject_empty_contents`?

Reply:
Which, again, as we discussed in the chat as well, would be confusing since contents such as `{ uint a = ... }` would also be considered "empty" here (as they didn't even match a prefix) and so this doesn't quite catch the meaning and could be confusing for the user.